.github/workflows/VersionCalPRComment.yml

```diff
@@ -0,0 +1,131 @@
+name: Comment PR new module release when pr committed and labeled/unlabeled
+
+on: 
+  workflow_dispatch:
+  pull_request_target:
+      types: [opened, labeled, unlabeled, synchronize]
+
+permissions:
+  pull-requests: write
+
+jobs:
+  version-cal:
+    if:  startsWith(github.head_ref, 'auto-version')
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Setup azdev
+        run: |
+          echo "message=$(echo 'Suggested init version: 1.0.0b1 for preview release and 1.0.0 for stable release')" >> $GITHUB_ENV
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v3
+        with:
+           python-version: "3.10"
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0 # checkout all branches
+      - name: Show workdirectory after site cloned
+        run: |
+           pwd
+           ls
+      - name: Get Diff Files
+        run: |
+            echo github.event.pull_request.base.sha: ${{github.event.pull_request.base.sha}}
+            echo github.event.pull_request.base.ref: ${{ github.event.pull_request.base.ref }}
+            echo github.sha: ${{github.sha}}
+            echo GITHUB_BASE_REF: ${GITHUB_BASE_REF}
+            echo GITHUB_HEAD_REF: ${GITHUB_HEAD_REF}
+            echo GITHUB_SHA: ${GITHUB_SHA}
+            echo GITHUB_REF: ${GITHUB_REF}
+            echo "base_branch=${GITHUB_BASE_REF}" >> $GITHUB_ENV
+            echo "diff_branch=${GITHUB_HEAD_REF}" >> $GITHUB_ENV
+            echo "base_commit=${GITHUB_SHA}" >> $GITHUB_ENV
+            echo "diff_commit=${GITHUB_SHA}" >> $GITHUB_ENV
+            git --version
+            git log --oneline | head -n 30
+            echo git branch -a
+            git branch -a
+            git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} > changed_files
+            cat changed_files
+            cat changed_files | grep azext_ | awk -F"azext_" '{print $1}'| awk -F"/" '{print $2}' | sort | uniq > changed_modules
+            echo "changed_module_list=$(cat changed_files | grep azext_ | awk -F"azext_" '{print $1}'| awk -F"/" '{print $2}' | sort | uniq | xargs)" >> $GITHUB_ENV
+      - name: Display Diff Modules
+        run: |
+            for mod in `cat changed_modules`
+            do
+              echo changed module: ${mod}
+            done
+      - uses: actions/checkout@master
+        with:
+           repository: Azure/azure-cli
+           path: ./azure-cli
```

Comment:
Why not checkout directly to the upper directory, but append a move operation?

Reply:
Because checkout do not allow path outside workdirectory, so cloned azure-cli first and then mv it parent path for azdev setup.

Reply:
This is the same as ADO :)